src/main/resources/schema/elytron-client-1_1.xsd

```diff
@@ -72,129 +122,413 @@
-            <xsd:element name="set-mechanism-realm" type="optional-name-type" minOccurs="0"/>
-            <xsd:element name="rewrite-user-name-regex" type="regex-substitution-type" minOccurs="0"/>
-            <xsd:element name="sasl-mechanism-selector" type="selector-type" minOccurs="0"/>
-            <xsd:element name="set-mechanism-properties" type="properties-type" minOccurs="0"/>
+            <xsd:element name="set-mechanism-realm" type="optional-name-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        The name of the realm that will be selected by the SASL mechanism if required.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="rewrite-user-name-regex" type="regex-substitution-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        A regular expression pattern and replacement to re-write the user name used for authentication.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="sasl-mechanism-selector" type="selector-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        A SASL mechanism selector using the syntax from org.wildfly.security.sasl.SaslMechanismSelector,fromString().
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="set-mechanism-properties" type="properties-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        One or more properties to be passed to the authentication mechanisms.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
             <xsd:element name="credentials" type="client-credentials-type" minOccurs="0"/>
-            <xsd:element name="set-authorization-name" type="name-type" minOccurs="0"/>
+            <xsd:element name="set-authorization-name" type="name-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        The name that should be used for authorization if different from the authentication identity.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
             <xsd:element name="providers" type="providers-type" minOccurs="0" />
             <xsd:element ref="abstract-sasl-factories" minOccurs="0"/>
         </xsd:all>
-        <xsd:attribute name="name" type="xsd:string" use="required"/>
+        <xsd:attribute name="name" type="xsd:string" use="required">
+        <xsd:annotation>
+            <xsd:documentation>
+                The name of the authentication configuration.
```

Comment:
Name used for referencing? (as already fixed for ssl-context - would be appropriate for probably all `<xsd:attribute name="name" type="xsd:string" use="required">` occurrences)